### DISCUSSION TITLE
Vert.x apiman 2.1.0

### DISCUSSION THREAD

**Discussion by potehvad** [apiman/apiman#1513] [Help]
Api request returns 431Request Header Fields Too Large

Any suggestions on:
1. what exact size of headers fields
2. how to enlarge it

**Comment by msavy** [comment]
You will need to provide a lot more information than that to get a useful answer

Which version of Apiman are you using
Which gateway are you using
Was it Apiman that triggered the 431 or the service behind Apiman
How big are the headers that you've set to cause that issue
How many headers did you set

**Comment by potehvad** [comment]
apiman version 2.1.0

As I have debugged error is triggered by vert.x or even with netty-transport component

Here is my request
`curl --location --request POST 'http://localhost:8080/coop/xs2a_payments/v1.3.1/payments' \
--header 'Content-Type: application/json' \
--header 'X-Request-ID: edeb1193-20cc-4499-b39e-237c0b94f023' \
--header 'X-API-Key: c9d51b93-a50f-40a1-9774-f2956394833a' \
--header 'Explicit-Authorisation-Preferred: false' \
--header 'Redirect-URI: http://www.google.com' \
--header 'Nok-Redirect-URI: http://www.google.com' \
--header 'IP-Address: 10.10.10.10' \
--header 'Signature-Certificate: MIIEZDCCA0ygAwIBAgIJAJ8sorcmkuauMA0GCSqGSIb3DQEBCwUAMDExLzAtBgNVBAMMJlRlc3QgQ0FPVT1UaWV0byBDYXJkc089VGlldG9MPVJpZ2FDPUxWMB4XDTIxMDcwMjA5NTY1OFoXDTMxMDYzMDA5NTY1OFowcDETMBEGA1UEAwwKZG9tYWluLmNvbTEPMA0GA1UECwwGMjIyMjIyMQ8wDQYDVQQKDAYyMjIyMjIxDjAMBgNVBAcMBUNlc2lzMQswCQYDVQQGEwJMVjEaMBgGA1UEYQwRUFNETFYtRkNNQy0yMjIyMjIwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQCDTLIS8+hAzoZMl5vHdI1oChroKk+9D3GOupG1jwqbonoGJ8IPjSXDkyWKtuaJxmSIVGjoPU11f2eqn2qdwh6FqQEPHWWhUO+rOXKudTpCyFi1lgEOP6GKuS7ApamFpE60ndyIc4LgnUUSz2dYIny82IDSCj3o0KkZU7M5uRg6m5GVoNrgOIFrKVxo3IK40pj85bbbWhmLfJbyn2EfhMVdrE9S0169G0FcH+yyQ6Yn6O71aehbksg4+jLMNNp2pc3LwT7LJ9XoMWhWFb4U/ETsFrNJutFWxTg81hu632OiSqsX8O0hnTayavUZrsaBCc3BmMqir1lctk/BZDrw9q7HAgMBAAGjggE+MIIBOjApBgNVHREEIjAggg1teS5kb21haW4uY29tgg95b3VyLmRvbWFpbi5jb20wEwYDVR0lBAwwCgYIKwYBBQUHAwIwgfcGCCsGAQUFBwEDBIHqMIHnMAgGBgQAjkYBATA3BgYEAI5GAQUwLTArFiBodHRwOi8vZXhhbXBsZS5vcmcvcGtpZGlzY2xvc3VyZRMHZXhhbXBsZTCBjAYGBACBmCcCMIGBMEwwEQYHBACBmCcBAwwGUFNQX0FJMBEGBwQAgZgnAQEMBlBTUF9BUzARBgcEAIGYJwECDAZQU1BfUEkwEQYHBACBmCcBBAwGUFNQX0lDDChGaW5hbmNpYWwgYW5kIENhcGl0YWwgTWFya2V0cyBDb21taXNzaW9uDAdMVi1GQ01DMBMGBgQAjkYBBjAJBgcEAI5GAQYDMA0GCSqGSIb3DQEBCwUAA4IBAQCRBaQEueP2GfeHD/g/qSg/pVZZ0GdhLwOrEhTDjgrjpR+rnSRrVdcy/rNRivfDWbqvTmVGWPp24UMkivVg2VoCWNg000AvOtZY6FG9eaPUF2hHwlyjQZ/Dkna0JYr9zREwPdEh4ul0h17JdcuF33omZ760ufVlIVptsjCXxXbi4eJhq6dfQrEligVc8jtoqZZhCMP4Nwi3/DpSQNqPjSmM1rk7L8mXnHytdAxNke8MRfFXlasj/ZQALzcfDaYWThgu1DSAUIAiWIEPV+ntE9A5Qs50Q7l5uFWBq+ClUef3KFJP3AybAKkFvS3DQRL72rfKN23QKqTXpP+4dmRcPZxN' \
--header 'Another-certificate: MIIEZDCCA0ygAwIBAgIJAJ8sorcmkuauMA0GCSqGSIb3DQEBCwUAMDExLzAtBgNVBAMMJlRlc3QgQ0FPVT1UaWV0byBDYXJkc089VGlldG9MPVJpZ2FDPUxWMB4XDTIxMDcwMjA5NTY1OFoXDTMxMDYzMDA5NTY1OFowcDETMBEGA1UEAwwKZG9tYWluLmNvbTEPMA0GA1UECwwGMjIyMjIyMQ8wDQYDVQQKDAYyMjIyMjIxDjAMBgNVBAcMBUNlc2lzMQswCQYDVQQGEwJMVjEaMBgGA1UEYQwRUFNETFYtRkNNQy0yMjIyMjIwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQCDTLIS8+hAzoZMl5vHdI1oChroKk+9D3GOupG1jwqbonoGJ8IPjSXDkyWKtuaJxmSIVGjoPU11f2eqn2qdwh6FqQEPHWWhUO+rOXKudTpCyFi1lgEOP6GKuS7ApamFpE60ndyIc4LgnUUSz2dYIny82IDSCj3o0KkZU7M5uRg6m5GVoNrgOIFrKVxo3IK40pj85bbbWhmLfJbyn2EfhMVdrE9S0169G0FcH+yyQ6Yn6O71aehbksg4+jLMNNp2pc3LwT7LJ9XoMWhWFb4U/ETsFrNJutFWxTg81hu632OiSqsX8O0hnTayavUZrsaBCc3BmMqir1lctk/BZDrw9q7HAgMBAAGjggE+MIIBOjApBgNVHREEIjAggg1teS5kb21haW4uY29tgg95b3VyLmRvbWFpbi5jb20wEwYDVR0lBAwwCgYIKwYBBQUHAwIwgfcGCCsGAQUFBwEDBIHqMIHnMAgGBgQAjkYBATA3BgYEAI5GAQUwLTArFiBodHRwOi8vZXhhbXBsZS5vcmcvcGtpZGlzY2xvc3VyZRMHZXhhbXBsZTCBjAYGBACBmCcCMIGBMEwwEQYHBACBmCcBAwwGUFNQX0FJMBEGBwQAgZgnAQEMBlBTUF9BUzARBgcEAIGYJwECDAZQU1BfUEkwEQYHBACBmCcBBAwGUFNQX0lDDChGaW5hbmNpYWwgYW5kIENhcGl0YWwgTWFya2V0cyBDb21taXNzaW9uDAdMVi1GQ01DMBMGBgQAjkYBBjAJBgcEAI5GAQYDMA0GCSqGSIb3DQEBCwUAA4IBAQCRBaQEueP2GfeHD/g/qSg/pVZZ0GdhLwOrEhTDjgrjpR+rnSRrVdcy/rNRivfDWbqvTmVGWPp24UMkivVg2VoCWNg000AvOtZY6FG9eaPUF2hHwlyjQZ/Dkna0JYr9zREwPdEh4ul0h17JdcuF33omZ760ufVlIVptsjCXxXbi4eJhq6dfQrEligVc8jtoqZZhCMP4Nwi3/DpSQNqPjSmM1rk7L8mXnHytdAxNke8MRfFXlasj/ZQALzcfDaYWThgu1DSAUIAiWIEPV+ntE9A5Qs50Q7l5uFWBq+ClUef3KFJP3AybAKkFvS3DQRL72rfKN23QKqTXpP+4dmRcPZxN' \
--header 'One-more-certificate: MIIEZDCCA0ygAwIBAgIJAJ8sorcmkuauMA0GCSqGSIb3DQEBCwUAMDExLzAtBgNVBAMMJlRlc3QgQ0FPVT1UaWV0byBDYXJkc089VGlldG9MPVJpZ2FDPUxWMB4XDTIxMDcwMjA5NTY1OFoXDTMxMDYzMDA5NTY1OFowcDETMBEGA1UEAwwKZG9tYWluLmNvbTEPMA0GA1UECwwGMjIyMjIyMQ8wDQYDVQQKDAYyMjIyMjIxDjAMBgNVBAcMBUNlc2lzMQswCQYDVQQGEwJMVjEaMBgGA1UEYQwRUFNETFYtRkNNQy0yMjIyMjIwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQCDTLIS8+hAzoZMl5vHdI1oChroKk+9D3GOupG1jwqbonoGJ8IPjSXDkyWKtuaJxmSIVGjoPU11f2eqn2qdwh6FqQEPHWWhUO+rOXKudTpCyFi1lgEOP6GKuS7ApamFpE60ndyIc4LgnUUSz2dYIny82IDSCj3o0KkZU7M5uRg6m5GVoNrgOIFrKVxo3IK40pj85bbbWhmLfJbyn2EfhMVdrE9S0169G0FcH+yyQ6Yn6O71aehbksg4+jLMNNp2pc3LwT7LJ9XoMWhWFb4U/ETsFrNJutFWxTg81hu632OiSqsX8O0hnTayavUZrsaBCc3BmMqir1lctk/BZDrw9q7HAgMBAAGjggE+MIIBOjApBgNVHREEIjAggg1teS5kb21haW4uY29tgg95b3VyLmRvbWFpbi5jb20wEwYDVR0lBAwwCgYIKwYBBQUHAwIwgfcGCCsGAQUFBwEDBIHqMIHnMAgGBgQAjkYBATA3BgYEAI5GAQUwLTArFiBodHRwOi8vZXhhbXBsZS5vcmcvcGtpZGlzY2xvc3VyZRMHZXhhbXBsZTCBjAYGBACBmCcCMIGBMEwwEQYHBACBmCcBAwwGUFNQX0FJMBEGBwQAgZgnAQEMBlBTUF9BUzARBgcEAIGYJwECDAZQU1BfUEkwEQYHBACBmCcBBAwGUFNQX0lDDChGaW5hbmNpYWwgYW5kIENhcGl0YWwgTWFya2V0cyBDb21taXNzaW9uDAdMVi1GQ01DMBMGBgQAjkYBBjAJBgcEAI5GAQYDMA0GCSqGSIb3DQEBCwUAA4IBAQCRBaQEueP2GfeHD/g/qSg/pVZZ0GdhLwOrEhTDjgrjpR+rnSRrVdcy/rNRivfDWbqvTmVGWPp24UMkivVg2VoCWNg000AvOtZY6FG9eaPUF2hHwlyjQZ/Dkna0JYr9zREwPdEh4ul0h17JdcuF33omZ760ufVlIVptsjCXxXbi4eJhq6dfQrEligVc8jtoqZZhCMP4Nwi3/DpSQNqPjSmM1rk7L8mXnHytdAxNke8MRfFXlasj/ZQALzcfDaYWThgu1DSAUIAiWIEPV+ntE9A5Qs50Q7l5uFWBq+ClUef3KFJP3AybAKkFvS3DQRL72rfKN23QKqTXpP+4dmRcPZxN' \
--header 'CLIENT_CERTIFICATE: MIIEZDCCA0ygAwIBAgIJAJ8sorcmkuauMA0GCSqGSIb3DQEBCwUAMDExLzAtBgNVBAMMJlRlc3QgQ0FPVT1UaWV0byBDYXJkc089VGlldG9MPVJpZ2FDPUxWMB4XDTIxMDcwMjA5NTY1OFoXDTMxMDYzMDA5NTY1OFowcDETMBEGA1UEAwwKZG9tYWluLmNvbTEPMA0GA1UECwwGMjIyMjIyMQ8wDQYDVQQKDAYyMjIyMjIxDjAMBgNVBAcMBUNlc2lzMQswCQYDVQQGEwJMVjEaMBgGA1UEYQwRUFNETFYtRkNNQy0yMjIyMjIwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQCDTLIS8+hAzoZMl5vHdI1oChroKk+9D3GOupG1jwqbonoGJ8IPjSXDkyWKtuaJxmSIVGjoPU11f2eqn2qdwh6FqQEPHWWhUO+rOXKudTpCyFi1lgEOP6GKuS7ApamFpE60ndyIc4LgnUUSz2dYIny82IDSCj3o0KkZU7M5uRg6m5GVoNrgOIFrKVxo3IK40pj85bbbWhmLfJbyn2EfhMVdrE9S0169G0FcH+yyQ6Yn6O71aehbksg4+jLMNNp2pc3LwT7LJ9XoMWhWFb4U/ETsFrNJutFWxTg81hu632OiSqsX8O0hnTayavUZrsaBCc3BmMqir1lctk/BZDrw9q7HAgMBAAGjggE+MIIBOjApBgNVHREEIjAggg1teS5kb21haW4uY29tgg95b3VyLmRvbWFpbi5jb20wEwYDVR0lBAwwCgYIKwYBBQUHAwIwgfcGCCsGAQUFBwEDBIHqMIHnMAgGBgQAjkYBATA3BgYEAI5GAQUwLTArFiBodHRwOi8vZXhhbXBsZS5vcmcvcGtpZGlzY2xvc3VyZRMHZXhhbXBsZTCBjAYGBACBmCcCMIGBMEwwEQYHBACBmCcBAwwGUFNQX0FJMBEGBwQAgZgnAQEMBlBTUF9BUzARBgcEAIGYJwECDAZQU1BfUEkwEQYHBACBmCcBBAwGUFNQX0lDDChGaW5hbmNpYWwgYW5kIENhcGl0YWwgTWFya2V0cyBDb21taXNzaW9uDAdMVi1GQ01DMBMGBgQAjkYBBjAJBgcEAI5GAQYDMA0GCSqGSIb3DQEBCwUAA4IBAQCRBaQEueP2GfeHD/g/qSg/pVZZ0GdhLwOrEhTDjgrjpR+rnSRrVdcy/rNRivfDWbqvTmVGWPp24UMkivVg2VoCWNg000AvOtZY6FG9eaPUF2hHwlyjQZ/Dkna0JYr9zREwPdEh4ul0h17JdcuF33omZ760ufVlIVptsjCXxXbi4eJhq6dfQrEligVc8jtoqZZhCMP4Nwi3/DpSQNqPjSmM1rk7L8mXnHytdAxNke8MRfFXlasj/ZQALzcfDaYWThgu1DSAUIAiWIEPV+ntE9A5Qs50Q7l5uFWBq+ClUef3KFJP3AybAKkFvS3DQRL72rfKN23QKqTXpP+4dmRcPZxN+BNp8S/Gy0sQxWI0z5hOyuXQesfEmaLo/2W52M/nxgkJWXeStwP1clMzKuNwwki9ouahXzNQr6Ow1rZuKh4iPbSLepoB/Ui7sDzJHMNhXPoEtopo1wo/IVpKdMg/MPBt7bl1YTNJVjZ3HY7htE8tx2ilqHDiybs+9H5PiXD4zDj+C05B8wBzzjzSSxhA6Hp2BAWvr2RRnFKW9yrP+BWxZyxGOhWuILCwi5kEd5D0J2S55JORUme1wFDqn6ksfz+n0ryyQFYcseL2WbKcrogBDH6GWQBwQfAbeRdv5oEGoFrJSW72xk3+lSFxRpNMvCikeeeDx5XWAbFTFw1sCAwEAAaOCA+cwggPjMAkGA1UdEwQCMAAwHwYDVR0jBBgwFoAU8cmEC9iy8daHkPP4WOqGsXMLKSowHQYDVR0OBBYEFKSgM2Y+2lR+OuYZ+tjTYkuEpNB/MA4GA1UdDwEB/wQEAwIFoDAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwVAYDVR0gBE0wSzA1BglghEIBGgEAAwMwKDAmBggrBgEFBQcCARYaaHR0cHM6Ly93d3cuYnV5cGFzcy5uby9jcHMwBwYFZ4EMAQEwCQYHBACL7EABBDBCBgNVHR8EOzA5MDegNaAzhjFodHRwOi8vY3JsLnRlc3Q0LmJ1eXBhc3Mubm8vY3JsL0JQQ2xhc3MzVDRDQTIuY3JsMBgGA1UdEQQRMA+CDWFwaS50aWV0by5jb20weAYIKwYBBQUHAQEEbDBqMCkGCCsGAQUFBzABhh1odHRwOi8vb2NzcC50ZXN0NC5idXlwYXNzLmNvbTA9BggrBgEFBQcwAoYxaHR0cDovL2NydC50ZXN0NC5idXlwYXNzLm5vL2NydC9CUENsYXNzM1Q0Q0EyLmNlcjCBtAYIKwYBBQUHAQMEgacwgaQwgYwGBgQAgZgnAjCBgTBMMBEGBwQAgZgnAQMMBlBTUF9BSTARBgcEAIGYJwEBDAZQU1BfQVMwEQYHBACBmCcBBAwGUFNQX0lDMBEGBwQAgZgnAQIMBlBTUF9QSQwoRmluYW5jaWFsIGFuZCBDYXBpdGFsIE1hcmtldHMgQ29tbWlzc2lvbgwHTFYtRkNNQzATBgYEAI5GAQYwCQYHBACORgEGAzCCAYAGCisGAQQB1nkCBAIEggFwBIIBbAFqAHcAsMyD5aX5fWuvfAnMKEkEhyrH6IsTLGNQt8b9JuFsbHcAAAFth3XKuwAABAMASDBGAiEAjL9IwZ54KgBOsKjMfrfCdPH9dZJUn/5sdEcOsZpKAKwCIQD9EAXs1VheWhG4ORZmbW9iigBFK5EZswru678p8d4iLwB2ALDMg+Wl+X1rr3wJzChJBIcqx+iLEyxjULfG/SbhbGx3AAABbYd1yrsAAAQDAEcwRQIhAJPxAqcrGNdG8HLRReMUCBp6K7OvflvVLmICPb0jzPzPAiBE/V7+D4zbt5i78SlgWqqzNk7dJVDY+ZM1zgaKjLJFdAB3ALDMg+Wl+X1rr3wJzChJBIcqx+iLEyxjULfG/SbhbGx3AAABbYd1yrsAAAQDAEgwRgIhAIGfJ4w5d3j96sFxWN2T18kihDJFvXKidcddIp3z/WmzAiEAntBJiek1PD4TAgAXgZ8khWBsvgLlBr41G8cdVHync1AwDQYJKoZIhvcNAQELBQADggEBAI9otIWu4Kg+WQO9AiHhP3hUAidXhxdG3WghHrvvlopNXq1v4DIKLf9ILkUDi/frhXO0FptVec3UuofD8WFaVUppfyGLv9BoQMtsWIKsnxunJSeUqus/yI3iccD5z+NJ+EUx3ug4EbCEMZ0hTDRiJu+Ldtjs6E4NNyv0LE58m1GXiDOQZ1d2ZTgKpfJkbRk78q40ZSpKtI/7IUACs1UnMGIuEdFUJkV818Kw3dwV2Ts3e7LX4DAcTQ7XxrbHdjghVa+NCxssD6Ti7mEuQQ7UhiesaQosp6MZQILEzrNnqCpP/HaT2w5zNvSy8OxtYangh/ypXmWoN2deBHtXZ/rO4bY=' \
--header 'User-ID: PSU001'`
